pkg/operator/operator.go

```diff
@@ -50,36 +50,39 @@ var (
 
 type csiSnapshotOperator struct {
 	client        OperatorClient
+	crdClient     apiextclient.Interface
 	versionGetter status.VersionGetter
 	recorder      events.Recorder
 }
 
 func NewCSISnapshotControllerOperator(
 	client OperatorClient,
+	crdClient apiextclient.Interface,
 	versionGetter status.VersionGetter,
 	recorder events.Recorder,
 ) operator.Runner {
 	csiOperator := &csiSnapshotOperator{
 		client:        client,
+		crdClient:     crdClient,
 		versionGetter: versionGetter,
 		recorder:      recorder,
 	}
+	csiOperator.initCRDs()
```

Comment:
It's good enough for a "first call", however, the operator should ensure that it re-creates the CRDs in case something removes it or changes is while the operator is running.

You should have an informer on CRDs and react on their changes.

See MCO:
https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/operator.go#L330 (syncRenderConfig re-creates / overwrites CRDs).